lld/MachO/ICF.cpp

```diff
@@ -263,6 +265,33 @@ void ICF::forEachClassRange(size_t begin, size_t end,
   }
 }
 
+// Find or create a symbol at offset 0 in the given section
+static Symbol *getThunkTargetSymbol(ConcatInputSection *isec) {
+  for (Symbol *sym : isec->symbols) {
+    if (auto *d = dyn_cast<Defined>(sym)) {
+      if (d->value == 0)
+        return sym;
+    }
+  }
+
+  std::string thunkName;
+  if (isec->symbols.size() == 0)
+    thunkName = isec->getName().str() + ".icf.0";
+  else
+    thunkName = isec->getName().str() + "icf.thunk.target" +
+                std::to_string(icfThunkCounter++);
+
+  // If no symbol found at offset 0, create one
+  auto *sym = make<Defined>(thunkName, /*file=*/nullptr, isec,
+                            /*value=*/0, /*size=*/isec->getSize(),
+                            /*isWeakDef=*/false, /*isExternal=*/false,
+                            /*isPrivateExtern=*/false, /*isThumb=*/false,
+                            /*isReferencedDynamically=*/false,
+                            /*noDeadStrip=*/false);
+  isec->symbols.push_back(sym);
```

Comment:
If we use `isec->symbols.insert(isec->symbols.begin(), sym)` then we will be able to locate the zero offset much more quickly, but of course it is more expensive than `push_back()`. So the perf depends on how often we are creating this symbol vs finding this symbol. Also, I'm not sure if there are correctness implications of using `insert()`.

Reply:
Another option is to maintain a map from `isec` to `sym`